### ISSUE TITLE
After release 0.6.4 polkadot project is stopped compiling.

### ISSUE THREAD

**Issue by Polkaverse** [boinkor-net/governor#245]
```console
error[E0308]: mismatched types
  --> /Users/pankaj/.cargo/git/checkouts/polkadot-sdk-cff69157b985ed76/3649614/substrate/client/rpc-servers/src/middleware/rate_limit.rs:43:76
   |
43 |             inner: Arc::new(RateLimitInner::direct_with_clock(Quota::per_minute(n), &clock)),
   |                             ---------------------------------                       ^^^^^^ expected `QuantaClock`, found `&QuantaClock`
   |                             |
   |                             arguments to this function are incorrect
   |
note: associated function defined here
  --> /Users/pankaj/.cargo/registry/src/index.crates.io-6f17d22bba15001f/governor-0.6.4/src/state/direct.rs:58:12
   |
58 |     pub fn direct_with_clock(quota: Quota, clock: C) -> Self {
   |            ^^^^^^^^^^^^^^^^^
help: consider removing the borrow
   |
43 -             inner: Arc::new(RateLimitInner::direct_with_clock(Quota::per_minute(n), &clock)),
43 +             inner: Arc::new(RateLimitInner::direct_with_clock(Quota::per_minute(n), clock)),
   |

For more information about this error, try `rustc --explain E0308`.
error: could not compile `sc-rpc-server` (lib) due to 1 previous error
```

**Comment by antifuchs** [comment]
Oh, oof. Sounds like the version should indicate it's an incompatible upgrade. Current thinking is that I'll try and undo that damage on the 0.6 track, then release the original change as 0.7.0.

**Comment by Polkaverse** [comment]
Great thanks @antifuchs

**Comment by antifuchs** [comment]
Worse yet, I thought this could get caught by cargo-semver-checks, but apparently it didn't: It says 0.6.4 is semver-compatible to 0.6.3. I didn't run it on this release, but that it's not catching this means it might not be worth relying on at all...

**Comment by antifuchs** [comment]
OK, update to this: A 0.6.5 release won't happen (annoyingly, _backing out_ the change is a semver-incompatible fix, and I'm not excited about turning off semver checks on the release mechanism after all).

I'll pull the 0.6.4 release, then cut a release for 0.7.0 with the exact same changes as were in 0.6.4.

**Comment by antifuchs** [comment]
OK, I've pulled 0.6.3 and released 0.7.0, which is basically 0.6.4 but with the semver bump. When you do upgrade, note that the QuantaClock can be cloned but you have to do it at the call site now, so you can do `RateLimiter.direct_with_clock(clock.clone())` 

Thanks for the report, I appreciate it!
